torch/utils/data/dataset.py

```diff
@@ -206,6 +208,54 @@ def __len__(self):
         return self.tensors[0].size(0)
 
 
+class StackDataset(Dataset[Union[Tuple[T_co, ...], Dict[str, T_co]]]):
```

Comment:
Another thing needs to call out. It might be better to define a `TypeVar` that contains either `Tuple[T_co, ...]` or `Dict[str, T_co]`.

Reply:
Not sure I get the idea.
Do you mean:

```
T_td = Union[Tuple[T_co, ...], Dict[str, T_co]]

class StackDataset(Dataset[T_td]):
```

instead of:

`class StackDataset(Dataset[Union[Tuple[T_co, ...], Dict[str, T_co]]]):`

Reply:
No. Let's do something like `T_stack = TypeVar('T_stack',  Tuple[T_co, ...], Dict[str, T_co])`. Using `Union` means the output can be either `Tuple` or `Dict`. But, using `TypeVar` is like template-style that only allows one type per dataset instance.

Reply:
ok. added.